src/datapilot/core/platforms/dbt/insights/checks/check_macro_has_desc.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+from datapilot.core.insights.utils import get_severity
+from datapilot.core.platforms.dbt.insights.checks.base import ChecksInsight
+from datapilot.core.platforms.dbt.insights.schema import DBTInsightResult
+from datapilot.core.platforms.dbt.insights.schema import DBTModelInsightResponse
+from datapilot.core.platforms.dbt.schemas.manifest import AltimateResourceType
+
+
+class CheckMacroHasDesc(ChecksInsight):
+    NAME = "Check Macro Has Desc"
+    ALIAS = "check_macro_has_desc"
+    DESCRIPTION = "Checks that macros have descriptions. Descriptions help in understanding the purpose of the macro."
+    REASON_TO_FLAG = "Macros without descriptions can lead to confusion and hinder effective data modeling and analysis. It's important to have descriptions for macros."
```

Comment:
Undocumented macros can cause misunderstandings and inefficiencies in data modeling and analysis, as they make it difficult to understand their purpose and usage. Clear descriptions are vital for accuracy and streamlined workflow.